client/app/scripts/superdesk-search/search.js

```diff
@@ -571,16 +571,14 @@
             return {
                 scope: {},
                 templateUrl: asset.templateUrl('superdesk-search/views/search-tags.html'),
-                //require: ['^sd-meta-terms'],
-                link: function(scope, elem, attr, sdMetaTermsCtrl) {
+                require: ['^sdMetaTerms'],
+                link: function(scope, elem, attr, controllers) {
 
-                    var update = function() {
-                        tags.initSelectedFacets().then(function(currentTags) {
-                            scope.tags = currentTags;
-                        });
-                    };
+                    var sdMetaTermsCtrl = controllers[0];
 
-                    update();
```

Comment:
Removed the `update()` function for better readability because it is not invoked anywhere else except when the directive is linked.